myia/compile/backends/python/python.py

```diff
@@ -197,6 +211,59 @@ def python_universe_getitem(c, universe, handle):
     return f"{universe}.get({handle})"
 
 
+def op_grad(f, *required_args):
+    """Implement operation grad using finite difference."""
+    from myia.debug.finite_diff import GradTester
+    import inspect
+
+    argnames = inspect.getfullargspec(f).args
+
+    def gradient(*args):
+        gt = GradTester(f, None, args, argnames)
+        ret = gt.compute_finite_diff()
+        out = []
+        for i in range(len(args)):
+            argname = argnames[i]
+            if not required_args or argname in required_args:
+                k = f"d{f.__name__}/d{argname}"
+                out.append(ret[k])
+        return out[0] if len(out) == 1 else out
+
+    return gradient
+
+
+def op_value_and_grad(f, *required_args):
+    """Implement operation value_and_grad."""
+    gradient = op_grad(f, *required_args)
+
+    def v_and_g(*args):
+        return f(*args), gradient(*args)
+
+    return v_and_g
+
+
+def op_user_switch(cond, if_true, if_false):
+    """Implementation for operation `user_switch`."""
+    return if_true if cond else if_false
+
+
+def op_range(start, stop=None, step=None):
+    """Implementation for operation range."""
+    from myia.operations.op_range import range_
+
+    return range_(start, stop, step)
```

Comment:
I think this may be better supported by implementing a `myia_iter/next/hasnext` function that can handle range (and possibly generalizes over Python's `iter/next`).